packages/eslint-plugin/docs/rules/return-await.mdx

```diff
@@ -260,3 +293,12 @@ async function validNever3() {
 
 </TabItem>
 </Tabs>
+
+### Options Summary
+
+|             Option              | Ordinary Context <br/> (stylistic preference) | Error Handling Context <br/> (catches subtle control flow bugs) |                       Should I use?                        |
```

Comment:
[Docs] I think this table is pretty useful. Proposal: how about moving this to within the `## Options` early on, to prime folks with it before they dive into the options?

I also think it would be useful to mention _in each option's section_ the pros & cons of that option. Right now, the `always` section just says _"Requires that all returned promises are `await`ed."_. It doesn't mention the option being preferred, or why it's preferred.

Reply:
> Proposal: how about moving this to within the `## Options` early on, to prime folks with it before they dive into the options?

Yeah, that was actually my first approach, but i was waffling on how to go about defining an error-handling context since that's required before the table... and I went with this approach out of laziness. I agree that putting it first would be better, though, so I'll give it another stab 🙂 

> I also think it would be useful to mention _in each option's section_ the pros & cons of that option. Right now, the `always` section just says _"Requires that all returned promises are `await`ed."_. It doesn't mention the option being preferred, or why it's preferred.

Feels like we're venturing into scope creep territory, but at the same time, we probably don't want this option to go out without good supporting docs 🤔 Will try to incorporate this 👍

Reply:
I made changes in this area. I definitely think the result is much better! Thanks for making the suggestions.